### PR TITLE
HIVE-25367: Fix TestReplicationScenariosAcidTables#testMultiDBTxn

### PR DESCRIPTION
**Pull request by hmangla98**
Fix TestReplicationScenariosAcidTables#testMultiDBTxn

Flaky test check is green.

http://ci.hive.apache.org/job/hive-flaky-check/379/

### CODE REVIEW

**Comment by pvary** [comment]
What was the root cause of the problem? Why does this change helps?

Thanks,
Peter

**Comment by hmangla98** [comment]
> What was the root cause of the problem? Why does this change helps?
> 
> Thanks,
> Peter

Runnable was being released prior to setting the task Error. If main thread calls result.gettaskError(), it would return null because it's not set at that time. If you introduce wait time after taskQueue.releaseRunnable(), you can reproduce this issue in local setup.

**Review by pvary — APPROVED** [review]

**Comment by pvary** [comment]
The change looks reasonable to me.
Thanks @hmangla98 for the explanation!

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java [hunk 1]
```diff
@@ -235,10 +235,7 @@ public WarehouseInstance run(String command) throws Throwable {
       driver.run(command);
       return this;
     } catch (CommandProcessorException e) {
-      if (e.getCause() != null) {
-        throw  e.getCause();
-      }
-      throw e;
+      throw e.getCause();
```
Comment (ayushtkn) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java:238]: we need not to revert this change right? this is just a null check..

File: ql/src/java/org/apache/hadoop/hive/ql/exec/TaskRunner.java [hunk 2]
```diff
@@ -110,10 +109,11 @@ public void runSequential() {
       LOG.error("Error in executeTask", t);
     }
     result.setExitVal(exitVal);
-    taskQueue.releaseRunnable();
     if (tsk.getException() != null) {
       result.setTaskError(tsk.getException());
     }
+    result.setRunning(false);
+    taskQueue.releaseRunnable();
```
Comment (ayushtkn) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/TaskRunner.java:116]: can we avoid all these {{result.setRunning(false)}} change and just pull {{result.setExitVal(exitVal)}} below {{result.setTaskError(tsk.getException());}}
I tried locally and it worked for me.

**Review by ayushtkn — CHANGES_REQUESTED** [review]

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java (hunk 1 above)
Comment (pvary) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java:238]: I would keep the null check, so next time we might have a little bit more info when there is an error

Comment (hmangla98) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java]: Reverted.

File: ql/src/java/org/apache/hadoop/hive/ql/exec/TaskRunner.java (hunk 2 above)
Comment (hmangla98) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/TaskRunner.java]: Done.

**Review by ayushtkn — APPROVED** [review]
